HW1/P2/P2.txt

```diff
@@ -1 +1,6 @@
 # Your discussion here
+My strategy to balance work load came from a consideration of the picture of the mandelbrot fractal. A lot of work seems to be done in the middle and on the axis of the fractal. Doing a simple cartesian product and naively partitioning is not a good design. Because it will sort of mesh the domain and some cell will have a lot more work which is very unbalanced as you can see in P2a_hist.
+
+My strategy was to rearange the bashes to balance the load using the modulo function. The implementation is that I gave a key to each batch which was the index of the pixel modulo 100 (because we have 100 partition). I am the slicing the domain and each batch has a 1-pixel large slice every 100 slice, so they allsome work from the work-intensive area. We see in P2b_hist.png that it is a winning strategy since now the work is balanced. Every worker does about 2 millions instructions.
```

Comment:
Yeah, I think you are right! In P2a, we used default partitioning. Just partition all the data into 100 partitions within fixed order, which is fast with the partition time. However, it does not balance the work, since some pixels are concentrated in some certain partitions, such as the light pixels are located in certain partitions, which cause the processing time for those partitions very slow.